executor/containerdexecutor/executor.go

```diff
@@ -152,8 +139,23 @@ func (w *containerdExecutor) Run(ctx context.Context, id string, root executor.M
 	}
 	defer namespace.Close()
 
+	hostNetMode := false
 	if meta.NetMode == pb.NetMode_HOST {
```

Comment:
I think this check in here does not cover all cases. Default net provider can also be in host for certain daemon configuration.

I think instead we should extend the `network.Provider` interface with `LocalhostAllowed()` method